parse-package-set/src/Main.purs

```diff
@@ -49,13 +53,15 @@ main :: Effect Unit
 main = runAff_ (either throwException mempty) do
   tmpPath <- liftEffect $ tmpdir "cst-integration-"
 
-  writeTextFile UTF8 (tmpPath <> "/spago.dhall") defaultSpagoDhall
+  liftEffect $ Console.log $ "Making new project in " <> tmpPath
 
-  let execOpts = Exec.defaultExecSyncOptions { cwd = Just tmpPath }
-  s <- liftEffect $ Buffer.toString UTF8 =<< Exec.execSync "spago ls packages" execOpts
-  let lines = Str.split (Str.Pattern "\n") s
-  let packages = Str.joinWith " " (String.takeWhile (_ /= ' ') <$> lines)
-  _ <- liftEffect $ Exec.execSync ("spago install " <> packages) execOpts
+  writeTextFile UTF8 (tmpPath <> "/spago.yaml") defaultSpagoYaml
+
+  s <- liftEffect $ Buffer.toString UTF8 =<< Exec.execSync' "spago ls packages --json" (_ { cwd = Just tmpPath })
+  packages <- case decodeJson =<< parseJson s of
+    Left err -> throwError $ error $ printJsonDecodeError err
+    Right (object :: Object Json) -> pure $ Object.keys object
```

Comment:
```suggestion
    Right object -> pure $ Object.keys object
```

Reply:
it will throw error `Object a0 is wrong type`